barman/cli.py

```diff
@@ -447,6 +489,20 @@ def backup(args):
             server.config.parallel_jobs = args.jobs
         if hasattr(args, "bwlimit"):
             server.config.bandwidth_limit = args.bwlimit
+        if args.compression_type is not None:
```

Comment:
As well as the validation of individual arguments there is a subsequent validation which needs to happen which is that the resulting `config` fields (which will be a combined result of these arguments and the configuration) combine to an allowed combination.

The PostgresBackupExecutor has a [validate_configuration](https://github.com/EnterpriseDB/barman/blob/9682b304e8ab2a344074d117d062bd668f8d095e/barman/backup_executor.py#L238) method so perhaps that is a good place to do such a thing. We could also have the executor set the backup_compression option explicitly if it is not set but `backup_compression_levell` is set and avoid the special case in BackupManager on L586.

Reply:
This was a nice idea but it won't work - validate_configuration gets called when the Server is created which is before the arguments are parsed. We would therefore need to put such validation somewhere else.